# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution

In [17]:
import pandas as pd
import numpy as np
import math
from scipy import stats

In [3]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [4]:
# number of callbacks for black-sounding names
sum(data[data.race=='w'].call)

235.0

In [14]:
# narrow down to columns of interest
data = pd.concat([data.race,data.call], axis=1)
print('sample size =',len(data))
data.head()

sample size = 4870


race  call
0    w   0.0
1    w   0.0
2    b   0.0
3    b   0.0
4    w   0.0

# Q1:- What test is appropriate for this problem? Does CLT apply?

Two sample t-test is appropriate here as we are trying to determine if two population means are equal or not.

There are two main conditions of CLT to be valid are :- 

one is Sample size should be large and observations must be independent of each other.
Now we checked our dataset has 4870 observations(> 30) which means we have enough large dataset and its given that the races are assigned randomly to the resumes when presented to the employer so it means oberservations are independent of each other.


# Q2:-What are the null and alternate hypotheses?

Null Hypothesis: the mean difference between callback rate for black & white-sounding names is zero

Alternative Hypothesis: there is a significant (non-zero) difference between callback rate for black & white-sounding names

In [6]:
w = data[data.race=='w']
b = data[data.race=='b']

# Q3:-Compute margin of error, confidence interval, and p-value.

In [15]:
# calculate statistics for binomial distributions
n1 = len(data[data['race'] == 'b']) # black sample size
n2 = len(data[data['race'] == 'w']) # white sample size
p1 = sum(data[data['race'] == 'b'].call)/n1 # black callback rate
p2 = sum(data[data['race'] == 'w'].call)/n2 # white callback rate

In [18]:
# conduct two-sample t-test
p_hat = (n1*p1+n2*p2)/(n1+n2) # joint callback
std_hat = math.sqrt(p_hat*(1-p_hat)*((1/n1)+(1/n2))) # joint standard deviation

test_stat = (p1-p2)/std_hat
p_val = stats.t.sf(abs(test_stat), df=len(data)-1)*2

print('test statistic:',test_stat)
print('p value:',p_val)

test statistic: -4.108412152434346
p value: 4.0493178875903686e-05


In [19]:

# calculate margins of error for the two distributions
stderr1 = math.sqrt((p1*(1-p1))/n1) # black standard error
stderr2 = math.sqrt((p2*(1-p2))/n2) # white standard error
stderr_diff = math.sqrt(math.pow(stderr1,2) + math.pow(stderr2,2)) # standard error for difference

# calculate confidence interval for difference in the distributions
conf_int_lower = abs(p1-p2)-1.96*stderr_diff
conf_int_upper = abs(p1-p2)+1.96*stderr_diff
conf_int = [conf_int_lower,conf_int_upper]

print('confidence interval for the difference of black & white callback rate:')
print(conf_int)

confidence interval for the difference of black & white callback rate:
[0.016777447859559147, 0.047288260559332024]


<div class="span5 alert alert-success">
<p> Your answers to Q4 and Q5 here </p>
</div>

# Q4:-Write a story describing the statistical significance in the context or the original problem.


Pvalue = 4.0493178875903686e-05 , confidence Interval= ( 0.01677, 0.047 ), 

pvalue is smaller than 0.05 significance level and so we will reject our null hypothesis and conclude that there is statistically significant difference between callback rate for black & white-sounding names.

# Q5:- Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

No, the conclusion above does not imply that race is the most important factor in callback success, because it solely examines race. All other relevant factors like education and years of experience would need to be examined and compared in order to identify the "most important" factor.